src/storage/indexed-db-storage.js

```diff
@@ -8,22 +8,39 @@ const DEFAULT_SATURN_STORAGE_NAME = 'saturn-client'
 
 /**
  * @function indexedDbStorage
- * @returns {import('./index.js').Storage}
+ * @returns {function():Promise<import('./index.js').Storage>}
  */
 export function indexedDbStorage () {
-  const indexedDbExists = (typeof self !== 'undefined') && self?.indexedDB
+  const indexedDbExists = typeof self !== 'undefined' && self?.indexedDB
   let dbPromise
-  if (indexedDbExists) {
-    dbPromise = openDB(DEFAULT_IDB_STORAGE_NAME, DEFAULT_IDB_VERSION, {
-      upgrade (db) {
-        db.createObjectStore(DEFAULT_SATURN_STORAGE_NAME)
-      }
-    })
-  }
 
-  return {
-    get: async (key) => indexedDbExists && (await dbPromise).get(DEFAULT_SATURN_STORAGE_NAME, key),
-    set: async (key, value) => indexedDbExists && (await dbPromise).put(DEFAULT_SATURN_STORAGE_NAME, value, key),
-    delete: async (key) => indexedDbExists && (await dbPromise).delete(DEFAULT_SATURN_STORAGE_NAME, key)
+  return async () => {
```

Comment:
I think this wrapping function may be unnecessary. Someone who wants to use `indexedDbStorage` can just pass the function reference to the js-client config, instead of calling it.

old:
```js
const prodSaturn = new Saturn({
    storage: indexedDbStorage(),
})
```

new:
```js
const prodSaturn = new Saturn({
    storage: indexedDbStorage,
})
```

Reply:
Yeah this makes sense and simplifies the process. 

This will require changes to the browser client to change the way the storage dependency is passed.

Reply:
yeah that's why I mentioned maintaining backwards compability with the config, something like

`this.storage = this.config.storage ?? await this.config.getStorage?.() ?? memoryStorage()`

Although we can just lock this update behind a minor version bump.